### DISCUSSION TITLE
how to add a button in the header navigation?

### DISCUSSION THREAD

**Discussion by carolhuang1990** [squidfunk/mkdocs-material#3008] [Q&A]
Hi,
Is there any way to add a button in the header navigation.
For example, I want to add a button [Live demo] beside the searching bar.
Is that possible?
I've tried to customize the header.html, but get some errors, and cannot find any instructions in the documentations.

Thanks&Regards
Carol

**Comment by linxianmiao** [comment]
class="md-button md-button--primary"
you can use <a> like this :
<a href="xxxx" class="md-button md-button--primary"></a>

**Reply by linxianmiao** [reply]
`<a href="xxxx" class="md-button md-button--primary"></a>`

**Reply by carolhuang1990** [reply]
@linxianmiao Thanks for your reply.
I just want to know how to add the button in the header (beside the searching box). :)
Or, how to customize the header.

**Comment by jasonren0403** [comment]
I think these docs will help:
* [Buttons](https://squidfunk.github.io/mkdocs-material/reference/buttons/)
* [Header - announcement bar](https://squidfunk.github.io/mkdocs-material/setup/setting-up-the-header/#announcement-bar)

That is to say, extend `base.html` and add button to block announce.

**Reply by linxianmiao** [reply]
black cat, my laopo!

**Comment by Stephen-Gates** [comment]
@carolhuang1990 did you get this working?

**Comment by LIGHTnik10** [comment]
Hi,
I know I am late but still want to just solution in case someone needs it in future.
**So to implement button in to the header you need to extend the theme**
[Extend i.e. Customize](https://squidfunk.github.io/mkdocs-material/customization/?h=exte#extending-the-theme)
1. Write overrides inside yml file.
2. Create a main.html file inside overrides.
3. Use this script(mentioned also in the link provided above):
`{% extends "base.html" %}

{% block scripts %}
  
  {{ super() }}
  
{% endblock %}
`
4.Before the super copy the header.html file code inside mkdocs-materail repo, just search for it you will find it and then edit that html by however you want the button to be. I usually take reference for buttons from w3 schools.
5. Run the file it should work.
I know it's not perfect, but it's what worked for me.
Feel free to drop in any queries.

**Reply by ooker777** [reply]
Here is the [header.html](https://github.com/squidfunk/mkdocs-material/blob/master/material/partials/header.html). To be clear, it's not copying the its content into `main.html`, but copy the file to your `overrides/partial/` folder

**Reply by ddxv** [reply]
Thanks, your posts helped me get it:

This doc is also very helpful and describes the general way to do this:
https://squidfunk.github.io/mkdocs-material/customization/?h=exte#extending-the-theme

I copied `header.html` from [header.html](https://github.com/squidfunk/mkdocs-material/blob/master/material/templates/partials/header.html) (note it seems to have moved since @ooker777's link) to `overrides/partials/header.html`

Inside `overrides/partials/header.html` I added:
```<a class="md-header__button" href="https://openattribution.dev"><h3>openattribution.dev</h3></a<>```

I used the md-header__button class as it's used for other items in the header.

**Comment by tpdorsey** [comment]
In a previous version of the Prefect docs, I added a tab menu item in the header bar with a link to our Cloud UI by extending the `tabs.html` partial: https://github.com/PrefectHQ/prefect/blob/2.8.4/docs/overrides/partials/tabs.html

Then to position it always on the right:

```
.cloud-link {
    margin-left: auto;
}
```

Maybe that gives you some ideas how to implement.

**Reply by ooker777** [reply]
It works. Thanks. Can you tell where to put that css?

**Reply by tpdorsey** [reply]
extra.css

**Comment by ooker777** [comment]
In my case I make use of the link to the git button in the header. It would look like this:

![](https://i.imgur.com/jetBZgH.png)

If you don't use a github link it won't display the tag/star/fork number below.

Put this in your `mkdocs.yml`:
```yaml
repo_url: https://example.com
repo_name: Button name
theme:
    icon:
      repo: octicons/download-16
```
